plugin/src/test/java/org/opensearch/ml/rest/SecureMLRestIT.java

```diff
@@ -194,4 +210,58 @@ public void testTrainModelInAsyncWayWithFullAccess() throws IOException {
         }, true);
     }
 
+    public void testReadOnlyUser_CanGetModel_CanNotDeleteModel() throws IOException {
+        KMeansParams kMeansParams = KMeansParams.builder().build();
+        // train model with full access client
+        train(mlFullAccessClient, FunctionName.KMEANS, irisIndex, kMeansParams, searchSourceBuilder, trainResult -> {
+            String modelId = (String) trainResult.get("model_id");
+            assertNotNull(modelId);
+            String status = (String) trainResult.get("status");
+            assertEquals(MLTaskState.COMPLETED.name(), status);
+            try {
+                // get model with readonly client
+                getModel(mlReadOnlyClient, modelId, model -> {
+                    String algorithm = (String) model.get("algorithm");
+                    assertEquals(FunctionName.KMEANS.name(), algorithm);
+                });
+            } catch (IOException e) {
+                assertNull(e);
+            }
+            try {
+                // Failed to delete model with read only client
+                deleteModel(mlReadOnlyClient, modelId, null);
+            } catch (Exception e) {
+                assertEquals(ResponseException.class, e.getClass());
+                assertTrue(Throwables.getStackTraceAsString(e).contains("no permissions for [cluster:admin/opensearch/ml/models/delete]"));
+            }
+        }, false);
+    }
+
+    public void testReadOnlyUser_CanGetTask_CanNotDeleteTask() throws IOException {
+        KMeansParams kMeansParams = KMeansParams.builder().build();
+        // train model with full access client
+        train(mlFullAccessClient, FunctionName.KMEANS, irisIndex, kMeansParams, searchSourceBuilder, trainResult -> {
+            assertFalse(trainResult.containsKey("model_id"));
+            String taskId = (String) trainResult.get("task_id");
+            assertNotNull(taskId);
+            String status = (String) trainResult.get("status");
+            assertEquals(MLTaskState.CREATED.name(), status);
+            try {
+                // get task with readonly client
+                getTask(mlFullAccessClient, taskId, task -> {
+                    String algorithm = (String) task.get("function_name");
+                    assertEquals(FunctionName.KMEANS.name(), algorithm);
+                });
+            } catch (IOException e) {
+                assertNull(e);
+            }
+            try {
+                // Failed to delete task with read only client
+                deleteTask(mlReadOnlyClient, taskId, null);
```

Comment:
Could this be risky test format since if no exception is thrown then it won't go into catch block and test will still pass. same for above delete model test

Reply:
Good catch, will add throwing exception after delete model/task. So if readonly user can delete model/task successfully, we will throw exception and that will make test fail.